### ISSUE TITLE
VanityUrls are inconsistent in a cluster

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#17278]
When running in a cluster, vanity urls can suddenly disappear and reappear, depending on the state of the cache.

We need to rewrite the VanityUrl cache completely, simplify it and load it eagerly

**Comment by jcastro-dotcms** [comment]
PR including fix from Support: https://github.com/dotCMS/core/pull/18378/files

**Comment by dsilvam** [comment]
PR: https://github.com/dotCMS/core/pull/18377

**Comment by nollymar** [comment]
**Note to QA:** These changes haven't been merged yet because VanityURLs need to be tested carefully before merging it.

**Comment by wezell** [comment]
@fmontes this needs internal QA

**Comment by DannyGloudemans** [comment]
Hi @wezell,

We are experiencing similar issues in dotCMS 5.2.6. We were able to reproduce this issue in both a single and clustered instance. We've tried to fix it with creating a hotfix plugin of the merge request: https://github.com/dotCMS/core/pull/18378/files.

However, this 'hotfix' will put the vanity URLs in a timed cache, in other words if the issue occurs it will only occur for X-seconds, however this doesn't resolve the actual issue or did we miss a part of the merge request to resolve the issue?

Thank you
